src/dmd/expressionsem.d

```diff
@@ -8836,15 +8836,6 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             exp = new ConstructExp(exp.loc, exp.e1, exp.e2);
             exp.type = t;
 
-            // @@@DEPRECATED_2020-06@@@
-            // When removing, alter `checkModifiable` to return the correct value.
```

Comment:
This comment doesn't seem to have been addressed

Reply:
I had a look, and I couldn't see the point of this comment.  Immutables are Modifiable.initialization, other kinds of variables are Modifiable.yes.  If you start returning `Modifiable.no` then you begin getting type related errors.

Reply:
Looking at the original PR, the flow is: `checkModifiable` => `checkModify` => `modifyFieldVar`.

It seems that `checkModifiable` returns true even if it errors out, relying on the side effects of `error`. I might have missed this while writing the comment (or perhaps wasn't precise enough, since the error is in `modifyFieldVar`, which is called by `checkModifiable`).

Reply:
A follow up to ensure that these functions are documented properly should be noted then, as I don't think it is written anywhere that `modifyFieldVar` returns `true` for initialization and `false` for yes, I just inferred that by reading the implementation/use.